### PR TITLE
Unify key generation self-signature

### PR DESCRIPTION
**Pull request by lubux**
- Serialized signature sub-packets are now ordered by their ID.
- The creation time, issuer key ID, and key flag sub-packets are now critical.

### CODE REVIEW

File: openpgp/key_generation.go [hunk 1]
```diff
@@ -151,7 +155,7 @@ func (t *Entity) addUserId(name, comment, email string, config *packet.Config, c
 
 	primary := t.PrivateKey
 	isPrimaryId := len(t.Identities) == 0
-	selfSignature := createSignaturePacket(&primary.PublicKey, packet.SigTypePositiveCert, config)
```
Comment (andrewgdotcom) [inline openpgp/key_generation.go:154]: What's the motivation behind this particular change? Many (most?) implementations use PositiveCert for self-signatures, and GenericCert for third-party signatures. I think this is a distinction worth preserving, even if it isn't universally followed...

Comment (andrewgdotcom) [reply openpgp/key_generation.go:154]: dkg referred to this in https://dkg.fifthhorseman.net/blog/gpg-ask-cert-level-considered-harmful.html

Comment (lubux) [reply openpgp/key_generation.go:154]: Hi Andrew. Thanks for your input. The goal of this merge request is to unify the key generation process with [openpgp.js](https://github.com/openpgpjs/openpgpjs), ensuring they produce the same self-signature during key generation. I don’t have a strong preference regarding the type of signature, as long as they are consistent.

@twiss, what’s your opinion on this?

Comment (twiss) [reply openpgp/key_generation.go:154]: Ah, I was not aware of that, I was under the impression that most use a generic cert. But yes, I'm fine with keeping a positive cert here and switching to that in OpenPGP.js instead :+1:

Comment (lubux) [reply openpgp/key_generation.go]: Reverted in: https://github.com/ProtonMail/go-crypto/pull/208/commits/20d98066bf49883040570c138d7a4edeba6dcdc7

**Comment by caarlos0** [comment]
see https://github.com/ProtonMail/go-crypto/issues/263

**Pull request merged**